### PR TITLE
chore: tests for BH behavior (department actions)

### PR DESCRIPTION
**Pull request by murtaza98**
## Further comments

Added tests for:
- https://rocketchat.atlassian.net/browse/OC-1079
- https://rocketchat.atlassian.net/browse/OC-1084
- https://rocketchat.atlassian.net/browse/OC-1086
- https://rocketchat.atlassian.net/browse/OC-1087

Note: These tests are built on top of https://github.com/RocketChat/Rocket.Chat/pull/29543 . This should only be merged after the aforementioned PR is merged

### CODE REVIEW

File: apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts [hunk 1]
```diff
@@ -286,4 +286,272 @@ describe('LIVECHAT - business hours', function () {
...
+	(IS_EE ? describe : describe.skip)('[EE] BH operations post department disablement', () => {
+		let defaultBusinessHour: ILivechatBusinessHour;
+		let customBusinessHour: ILivechatBusinessHour;
+		let deptLinkedToCustomBH: ILivechatDepartment;
+		let agentLinkedToDept: Awaited<ReturnType<typeof createDepartmentWithAnOnlineAgent>>['agent'];
+
+		before(async () => {
+			await updateSetting('Livechat_business_hour_type', LivechatBusinessHourBehaviors.MULTIPLE);
+			// wait for the callbacks to be registered
+			await sleep(2000);
```
Comment (KevLehman) [inline apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts:298]: Wasn't this 1000 in other PR?

File: apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts [hunk 2]
```diff
@@ -286,4 +286,272 @@ describe('LIVECHAT - business hours', function () {
...
+
+			// verify if BH link is removed
+			const department = await getDepartmentById(deptLinkedToCustomBH._id);
+			expect(department).to.be.an('object');
+			expect(department.businessHourId).to.be.undefined;
+
+			// verify if BH is closed
+			const latestCustomBH = await getCustomBusinessHourById(customBusinessHour._id);
+			expect(latestCustomBH).to.be.an('object');
+			expect(latestCustomBH.active).to.be.false;
```
Comment (KevLehman) [inline apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts:333]: You can also use `to.have.property('businessHourId', false)`

File: apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts [hunk 3]
```diff
@@ -286,4 +286,272 @@ describe('LIVECHAT - business hours', function () {
...
+
+			await deleteDepartment(deptLinkedToCustomBH._id);
+
+			const latestAgent: ILivechatAgent = await getMe(agentLinkedToDept.credentials as any);
+			expect(latestAgent).to.be.an('object');
+			expect(latestAgent.openBusinessHours).to.be.an('array').of.length(1);
+			expect(latestAgent?.openBusinessHours?.[0]).to.be.equal(defaultBusinessHour._id);
+		});
+
+		it('upon deleting a department, overlapping agents should still have BH within their cache', async () => {
```
Comment (KevLehman) [inline apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts:511]: same for this test

**Review by KevLehman — DISMISSED** [review]

File: apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts [hunk 4]
```diff
@@ -286,4 +286,272 @@ describe('LIVECHAT - business hours', function () {
...
+			// disable department
+			await disableDepartment(deptLinkedToCustomBH);
+
+			const latestAgent: ILivechatAgent = await getMe(agentLinkedToDept.credentials as any);
+			expect(latestAgent).to.be.an('object');
+			expect(latestAgent.openBusinessHours).to.be.an('array').of.length(1);
+			expect(latestAgent?.openBusinessHours?.[0]).to.be.equal(defaultBusinessHour._id);
+		});
+
+		it('upon disabling a department, overlapping agents should still have BH within their cache', async () => {
```
[the remark this answers is not in the document]
Comment (murtaza98) [reply apps/meteor/tests/end-to-end/api/livechat/19-business-hours.ts:496]: Okay I've reduced the timeout and hope it shortens the duration

**Review by KevLehman — APPROVED** [review]

**Pull request merged**